src/main/java/org/alfresco/opencmis/AlfrescoCmisServiceImpl.java

```diff
@@ -1526,15 +1527,32 @@ public void appendContentStream(String repositoryId, Holder<String> objectId, Ho
             throw new CmisStreamNotSupportedException("Document type doesn't allow content!");
         }
 
-        try
-        {
-        	connector.appendContent(info, contentStream, isLastChunk);
-            objectId.setValue(connector.createObjectId(nodeRef));
-        }
-    	catch(IOException e)
-    	{
-    		throw new ContentIOException("", e);
-    	}
+        //REPO-2182 - Separated appendContent and objectId.setValue in two different transactions because
+        //after executing appendContent, the new objectId is not visible.
+        RetryingTransactionHelper helper = connector.getRetryingTransactionHelper();
+        helper.doInTransaction(new RetryingTransactionCallback<Void>()
+        {
+           public Void execute() throws Throwable
+           {
+               try
+               {
+                   connector.appendContent(info, contentStream, isLastChunk);
+               }
+               catch(IOException e)
+               {
+                    throw new ContentIOException("", e);
+               }
+                return null;
+            }}, false, true);
+
+       String objId = helper.doInTransaction(new RetryingTransactionCallback<String>()

```

Comment:
This second transaction is independent of the first inner transaction; we've lost atomicity of the overall call.